lib/Backend/FlowGraph.cpp

```diff
@@ -67,13 +82,19 @@ bool FlowGraph::Dominates(Region *region1, Region *region2)
 bool FlowGraph::DoesExitLabelDominate(IR::BranchInstr *leaveInstr)
 {
     IR::LabelInstr * leaveTarget = leaveInstr->GetTarget();
-    Assert(leaveTarget->GetNextRealInstr()->IsBranchInstr());
-    IR::BranchInstr *leaveChain = leaveTarget->GetNextRealInstr()->AsBranchInstr();
+    Assert(leaveTarget->GetNextRealInstr()->IsBranchInstr() || leaveTarget->GetNextRealInstr()->m_opcode == Js::OpCode::ProfiledLoopEnd);
+    IR::BranchInstr *leaveChain = leaveTarget->GetNextBranchOrLabel()->AsBranchInstr();
```

Comment:
Won't the `AsBranchInstr()` fail in the case when `leaveTarget->GetNextRealInstr()->m_opcode == Js::OpCode::ProfiledLoopEnd`?

Reply:
GetNextBranchOrLabel() will skip over the ProfiledLoopEnd instruction

Reply:
General pattern in case of early returns look like this : 
```
for(...) {
  try {
     if (cond) {
       return;
    }
  }
 finally {
}
}
```

Reply:
Ahh, that makes sense.

Reply:
The IR looks like this more or less : 
```
$loopTop: ProfiledLoopBodyStart
...
	TryFinally  $finallyLabel
	BrGt_A      $earlyExit, <condition>
	Br          $normalExit
$earlyExit:
    s0.var          =  Ld_A           Undefined
                       Leave          $L2
$L2:                                     
                       ProfiledLoopEnd
                       Br             $exit
$normalExit:
                       Leave          $L3 
$L3:                                      
                       Br             $L5 
$finallyLabel:                                      
                       Finally            
                       LeaveNull          
$L5:                                      
                       <loopIncr>
                       Br $loopTop     
$L10:                                 
                       ProfiledLoopEnd
    s0.var          =  Ld_A           0xXXXXXXXX (unknown)[Undefined].var
$exit:                                                                   
                       Ret            s0.var                   
```